src/app/PageWrapper.ts

```diff
@@ -166,6 +166,17 @@ export class PageWrapper {
     await this.page.hover(elementId);
     console.log("Hovered: ", elementId);
   }
+
+  /**
+   * function to emulate network traffic 
+   * pass offline : true for making page offline
+   */
+   async emulateNetwork(offline: boolean, latency: number, downloadThroughput: number, uploadThroughput: number){
```

Comment:
instead of exposing all the params, let's keep this function private and expose - 
`setInternetEnabled(enabled: bool)` which can be called from tests. this will look more cleaner, the -1 being passed in tests at multiple places is a bit confusing.

```
page.setInternetEnabled(false); // for offline
page.setInternetEnabled(true); // for online
```